projects/gateway/pkg/translator/route_converter.go

```diff
@@ -88,9 +95,22 @@ func (rv *routeVisitor) ConvertRoute(gatewayRoute *gatewayv1.Route) ([]*gloov1.R
 		matchers = gatewayRoute.Matchers
 	}
 
+	routeDisplayName := gatewayRoute.Name
+	if routeDisplayName == "" {
+		routeDisplayName = "<unnamed>"
+	} else {
+		rv.containsNamedRoute = true
+	}
+	rv.nameTree += "_route:" + routeDisplayName
 	glooRoute := &gloov1.Route{
 		Matchers: matchers,
 		Options:  gatewayRoute.Options,
+		Name:     rv.nameTree,
+	}
+
+	// if this is a leaf and there are no named routes in the tree, wipe the name
+	if _, ok := gatewayRoute.Action.(*gatewayv1.Route_DelegateAction); !ok && !rv.containsNamedRoute {
```

Comment:
Nit: this can be simplified like this:
```suggestion
	if gatewayRoute.GetDelegateAction() == nil && !rv.containsNamedRoute {
```